libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -32,43 +34,68 @@ export class DefaultConfigurationSettingsConverter implements ConfigurationSetti
-      SetConfigurationSettingParam<string | FeatureFlagValue>
+      SetConfigurationSettingParam
     >();
-    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
-    let foundFeatureManagement = false;
-    let featureFlags: any = {};
-    let featureManagementKeyWord = "";
-    let enabledForKeyWord = "";
+
+    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam>();
+    let foundMsFmSchema = false;
+    let foundLegacySchema = false;
+    let msFmFeatureManagementKeyWord = "";
+    let legacySchemaFeatureManagementKeyWord = "";
+    let legacySchemaEnabledForKeyWord = "";
+    let legachSchemaRequirementTypeKeyWord = "";
+
+    const featureFlagsDict: any = {};
 
     if (options.format == ConfigurationFormat.Properties && !options.skipFeatureFlags) {
       this.checkFeatureManagementExist(config);
     }
 
-    for (let i = 0; i < Constants.FeatureManagementKeyWords.length; i++) {
+    for (let i = 0; i < Constants.FeatureManagementKeyWords.length - 1; i++) {
       if (Constants.FeatureManagementKeyWords[i] in config) {
-        if (foundFeatureManagement) {
+        if (foundLegacySchema) {
           throw new ArgumentError(
-            "Unable to proceed because data contains multiple sections corresponding to Feature Management."
+            `Unable to proceed because data contains multiple sections corresponding to Feature Management. with the key, ${Constants.FeatureManagementKeyWords[i]}`
           );
         }
+        foundLegacySchema = true;
+
+        legacySchemaFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
+        legacySchemaEnabledForKeyWord = Constants.EnabledForKeyWords[i];
+        legachSchemaRequirementTypeKeyWord = Constants.RequirementTypeKeyWords[i];
 
-        featureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
-        enabledForKeyWord = Constants.EnabledForKeyWords[i];
-        const featureManagementKey = featureManagementKeyWord as keyof object;
-        featureFlags = config[featureManagementKey];
-        delete config[featureManagementKey];
-        foundFeatureManagement = true;
+        const legacyFeatureManagementKey = legacySchemaFeatureManagementKeyWord as keyof object;
+        featureFlagsDict[legacyFeatureManagementKey] = config[legacyFeatureManagementKey];
+        delete config[legacyFeatureManagementKey];
       }
     }
 
-    if (foundFeatureManagement && !options.skipFeatureFlags) {
+    const msFmKeyWord = Constants.FeatureManagementKeyWords[3];
+    if (msFmKeyWord in config) {
+      if (foundLegacySchema) {
+        if (Object.keys(config[msFmKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord)) {
+          throw new ArgumentError(
+            `Unable to proceed because data contains an already defined section with same schema corresponding to Feature Management with the key, ${legacySchemaFeatureManagementKeyWord}.`
+          );
+        }
```

Comment:
Do we have a check to verify if `feature_management` section is present we have `feature_flags` array declared ?

Reply:
We don't need that, because old schema per se allows using the underscore naming style. Means we should still treat this as a valid scenario:

```
{
   "feature_management": {
        "FeatureA": true
  }
}
```